mantidproject/mantid
# ISIS Reflectometry (Polref) interface: Implemented uniform time slicing options in 'Event Handling' tab

This PR adds two new slicing options to the `Event Handling` tab in the interface called`'UniformEven` and `Uniform` respectively. A third option `LogValue` has been added, but is currently non-functional and will be implemented in another issue.

**To test:**

- If using windows, this must be tested in Release mode.
- The interface may be accessed via Interfaces->Reflectometry->ISIS Reflectometry (Polref). From there, go to the `Event Handling` tab.
- Each option can be separately activated by clicking on its radio button. Ensure that only one option is activated at any time.
- Go to the `Runs` tab and load `INTER_NR_test2.tbl` (from the sample INTER dataset) and click on `Process`. This should reduce the table and produce several workspaces. Ensure that these are the same as committing the same process on master.
- In `Event Handling`, select `Custom slicing`  and enter values `0, 10, 20, 30`. Go back to `Runs`, select the first group and process. This should produce sliced workspaces with time ranges of `0-10`, `10-20` and `20-30` seconds respectively. Ensure that the results match that of master (although the workspaces will be named by slice index, rather than time range).
- Now select `Uniform even slicing` and enter `5`. Go back to `Runs`, select the first group and process. This should  produce 5 evenly-sized sliced workspaces of each type.
- Finally select `Uniform slicing` and enter `10`. Runs `13460` and `13460` have a duration of 52 and 21 seconds respectively. Therefore this should produce 6 and 3 sliced workspaces of each type for each run. 
As only common slices are used for the stitched IvsQ workspaces, there should only be three sliced workspaces for them.
- Open a new table and add run `38415` in the first row, with `38393` as transmission run. Process with each slicing option and ensure that the `plot rows` and `plot group` buttons work correctly and produce the desired result.
- Repeat previous step (using one of the slicing options) and delete `TOF_38415` from the workspaces dock. Ensure that when attempting to plot, the interface gives an error message.
- Ensure documentation is accurate and up to date.

Fixes #18500.

*Release notes have been updated*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
